analyzers/src/SonarAnalyzer.CSharp/Rules/StaticFieldInGenericClass.cs

```diff
@@ -18,59 +18,65 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
-namespace SonarAnalyzer.Rules.CSharp
+namespace SonarAnalyzer.Rules.CSharp;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class StaticFieldInGenericClass : SonarDiagnosticAnalyzer
 {
-    [DiagnosticAnalyzer(LanguageNames.CSharp)]
-    public sealed class StaticFieldInGenericClass : SonarDiagnosticAnalyzer
-    {
-        private const string DiagnosticId = "S2743";
-        private const string MessageFormat = "A static field in a generic type is not shared among instances of different close constructed types.";
+    private const string DiagnosticId = "S2743";
+    private const string MessageFormat = "A static field in a generic type is not shared among instances of different close constructed types.";
 
-        private static readonly DiagnosticDescriptor Rule = DescriptorFactory.Create(DiagnosticId, MessageFormat);
+    private static readonly DiagnosticDescriptor Rule = DescriptorFactory.Create(DiagnosticId, MessageFormat);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
 
-        protected override void Initialize(SonarAnalysisContext context) =>
-            context.RegisterNodeAction(c =>
+    protected override void Initialize(SonarAnalysisContext context) =>
+        context.RegisterNodeAction(c =>
+            {
+                var typeDeclaration = (TypeDeclarationSyntax)c.Node;
+                if (c.IsRedundantPositionalRecordContext()
+                    || typeDeclaration.TypeParameterList is null
+                    || typeDeclaration.TypeParameterList.Parameters.Count < 1)
+                {
+                    return;
+                }
+                var typeParameterNames = typeDeclaration.TypeParameterList.Parameters.Select(x => x.Identifier.ToString()).ToArray();
+                if (typeDeclaration.BaseList is not null && typeDeclaration.BaseList.Types.Any(x => x.Type is GenericNameSyntax genericType && HasGenericTypeArgument(genericType, typeParameterNames)))
```

Comment:
```suggestion
                if (typeDeclaration.BaseList is {} baselist && baseList.Types.Any(x => x.Type is GenericNameSyntax genericType && HasGenericTypeArgument(genericType, typeParameterNames)))
```

I'd also move this condition into a local function and just add it as an extract condition to like 41.

Reply:
I'll fix this in #9405 as it moves the line around anyway